crates/sputnik/src/session.rs

```diff
@@ -0,0 +1,130 @@
+use ci_info::types::Vendor as CiVendor;
+use reqwest::Url;
+use semver::Version;
+use serde::Serialize;
+use sha2::{Digest, Sha256};
+use uuid::Uuid;
+
+use std::env::{consts::OS, current_dir};
+use std::fmt::Debug;
+use std::time::Duration;
+
+use crate::{Report, SputnikError};
+
+/// The Session represents a usage of the CLI analogous to a web session
+/// It contains the "url" (command path + flags) but doesn't contain any
+/// values entered by the user. It also contains some identity information
+/// for the user
+#[derive(Debug, Serialize)]
+pub struct Session {
+    /// the command usage where commands are paths and flags are query strings
+    /// i.e. rover profile list --help would become rover/profile/list?help
```

Comment:
I think this was my original comment and we didn't end up doing this so we should change this comment :)